src/grpc/search.rs

```diff
@@ -95,70 +99,209 @@ impl SearchService for SearchServiceImpl {
-        let token = tonic_auth!(
-            get_token_from_md(&request_metadata),
-            "Token extraction failed"
-        );
-
         // Validate format of provided id
         let resource_ulid = tonic_invalid!(
             DieselUlid::from_str(&inner_request.resource_id),
             "Invalid resource id format"
         );
 
-        // Check permissions
-        let ctx = Context::res_ctx(resource_ulid, DbPermissionLevel::READ, true);
-        tonic_auth!(
-            self.authorizer.check_permissions(&token, vec![ctx]).await,
-            "Permission denied"
-        );
+        let (object_plus, permission) = if request_metadata.get("Authorization").is_some() {
+            // Extract token and check permissions with empty context
+            let token = tonic_auth!(
+                get_token_from_md(&request_metadata),
+                "Token extraction failed"
+            );
 
-        // Get Object from cache
-        let mut object_plus = self
-            .cache
-            .get_object(&resource_ulid)
-            .ok_or_else(|| Status::not_found("Object not found"))?;
+            // Check permissions
+            let ctx = Context::res_ctx(resource_ulid, DbPermissionLevel::READ, true);
+            let user = tonic_auth!(
+                self.authorizer.check_permissions(&token, vec![ctx]).await,
+                "Permission denied"
+            );
+            let object = self
+                .cache
+                .get_object(&resource_ulid)
+                .ok_or_else(|| Status::not_found("Object not found"))?;
+            let mapping_perm = *self
+                .cache
+                .get_user(&user)
+                .ok_or_else(|| Status::not_found("User not found"))?
+                .attributes
+                .0
+                .permissions
+                .get(&resource_ulid)
+                .ok_or_else(|| Status::not_found("No permissions found"))?;
+            let permission = match mapping_perm {
+                ObjectMapping::OBJECT(perm) => perm.into(),
+                ObjectMapping::COLLECTION(perm) => perm.into(),
+                ObjectMapping::DATASET(perm) => perm.into(),
+                ObjectMapping::PROJECT(perm) => perm.into(),
+            };
+            (object, permission)
+        } else {
+            // Get Object from cache
+            let mut object_plus = self
+                .cache
+                .get_object(&resource_ulid)
+                .ok_or_else(|| Status::not_found("Object not found"))?;
 
-        // Check if object metadata is publicly available
-        match object_plus.object.data_class {
-            DataClass::PUBLIC | DataClass::PRIVATE => {}
-            _ => return Err(Status::invalid_argument("Resource is not public")),
-        }
+            // Check if object metadata is publicly available
+            match object_plus.object.data_class {
+                DataClass::PUBLIC => {}
```

Comment:
Private objects should get (redacted) infos as well. So it was correct before, but ultimately we should tie this to the "Metadata" License.